gunpowder/nodes/wkw_source.py

```diff
@@ -0,0 +1,188 @@
+import logging
+import numpy as np
+
+from gunpowder.batch import Batch
+from gunpowder.coordinate import Coordinate
+from gunpowder.profiling import Timing
+from gunpowder.roi import Roi
+from gunpowder.array import Array
+from gunpowder.array_spec import ArraySpec
+from .batch_provider import BatchProvider
+
+from webknossos import Dataset, Mag
+from webknossos.dataset.properties import (
+    _properties_floating_type_to_python_type
+)
+
+logger = logging.getLogger(__name__)
+
+
+class WKWSource(BatchProvider):
+    def __init__(
+            self,
+            filename,
+            datasets,
+            array_specs=None,
+            mag_specs=None,
```

Comment:
Sure, I can add a docstring.

Just for better understanding:
WKW stores different `magnifications`. So the raw microscope data is stored at magnification 1-1-1 ( let's say with 8 x 8 x 35 nm^3 voxel size), additional downsampled versions can be stored in the same dataset. For example 2-2-1 and 4-4-1 with voxel sizes 16 x 16 x 35 and 32 x 32 x 35, respectively.

In an optimal case, `WKWSource` would wait for the request voxel size and return the corresponding magnification on  the fly (or downsample an existing `magnification` to match the request). Since this is the first attempt to write a `gunpowder` batch provider, I decided for an initialization parameter for each `ArrayKey`. The type for `mag_specs` would be:
`Dict[ArrayKey, Union[str, List[int]]]`, i.e. `{raw: "4-4-1"}`

Reply:
Oh Ok. Thanks!
So far there hasn't been much of an attempt to make the pipeline dynamically fulfill requested voxel_size, dtype, etc.
It shouldn't be too difficult but I guess there hasn't been a good use case for it yet. You can still request multiple resolutions fairly easily with something like `mag_specs={raw_fullres: "1-1-1", raw_lowres: "4-4-1"}`